presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcSelectivePageSourceFactory.java

```diff
@@ -597,6 +598,11 @@ private static OrcPredicate toOrcPredicate(TupleDomain<Subfield> domainPredicate
             return filterFunctions.build();
         }
 
+        if (!isAdaptiveFilterReorderingEnabled(session)) {
+            filterFunctions.add(new FilterFunction(session, determinismEvaluator.isDeterministic(filter), predicateCompiler.compilePredicate(session.getSqlFunctionProperties(), filter).get()));
```

Comment:
Might be a coincidence that the code block inside ` if (conjuncts.size() == 1) {` below is the exact same as this.
If it is not try refactoring this into a new method to avoid duplication.

Reply:
@ajaygeorge In general, we favor inlining over tiny methods to reduce the need for jumping around when reading code.